### PR TITLE
Report build status to telemetry and segment

### PR DESCRIPTION
**Pull request by ferozemohideen**
## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Other (please describe):

## Pull request checklist

Please check if your PR fulfills the following requirements:

- [ ] If it's a backend change, tests for the changes have been added and `go test ./...` runs successfully from the root folder.
- [ ] If it's a frontend change, Prettier has been run
- [ ] Docs have been reviewed and added / updated if needed

### CODE REVIEW

File: api/server/handlers/porter_app/create_and_update_events.go [hunk 1]
```diff
@@ -62,6 +65,14 @@ func (p *CreateUpdatePorterAppEventHandler) ServeHTTP(w http.ResponseWriter, r *
 		telemetry.AttributeKV{Key: "porter-app-event-id", Value: request.ID},
 	)
 
+	if string(request.Type) == string(types.PorterAppEventType_Build) {
```
Comment (stefanmcshane) [inline api/server/handlers/porter_app/create_and_update_events.go:68]: Are these both not strings by default?

File: api/server/handlers/porter_app/create_and_update_events.go [hunk 2]
```diff
@@ -36,6 +37,8 @@ func (p *CreateUpdatePorterAppEventHandler) ServeHTTP(w http.ResponseWriter, r *
 	defer span.End()
 
 	cluster, _ := ctx.Value(types.ClusterScope).(*models.Cluster)
+	user, _ := ctx.Value(types.UserScope).(*models.User)
+	project, _ := ctx.Value(types.ProjectScope).(*models.Project)
 	telemetry.WithAttributes(span,
 		telemetry.AttributeKV{Key: "cluster-id", Value: int(cluster.ID)},
```
Comment (stefanmcshane) [inline api/server/handlers/porter_app/create_and_update_events.go:43]: Can drop these also

File: api/server/handlers/porter_app/create_and_update_events.go [hunk 3]
```diff
@@ -62,6 +65,14 @@ func (p *CreateUpdatePorterAppEventHandler) ServeHTTP(w http.ResponseWriter, r *
 		telemetry.AttributeKV{Key: "porter-app-event-id", Value: request.ID},
 	)
 
+	if string(request.Type) == string(types.PorterAppEventType_Build) {
+		if errors, ok := request.Metadata["errors"]; ok {
+			if errs, ok := errors.(map[string]error); ok {
```
Comment (stefanmcshane) [inline api/server/handlers/porter_app/create_and_update_events.go:70]: Have you tested that this wont panic?

File: api/server/handlers/porter_app/create_and_update_events.go [hunk 4]
```diff
@@ -82,6 +93,18 @@ func (p *CreateUpdatePorterAppEventHandler) ServeHTTP(w http.ResponseWriter, r *
 	p.WriteResult(w, r, event)
 }
 
+func reportErrors(ctx context.Context, errs map[string]error, config *config.Config, user *models.User, project *models.Project, stackName string) {
+	ctx, span := telemetry.NewSpan(ctx, "report-build-errors")
+	defer span.End()
+	var errStr string
+	for k, v := range errs {
```
Comment (stefanmcshane) [inline api/server/handlers/porter_app/create_and_update_events.go:100]: {`resource-k`: `v`}

File: api/server/handlers/porter_app/create_and_update_events.go (hunk 1 above)
Comment (ferozemohideen) [reply api/server/handlers/porter_app/create_and_update_events.go:68]: they're both PorterAppEventType so I can get rid of the string casting actually

File: api/server/handlers/porter_app/create_and_update_events.go (hunk 3 above)
Comment (ferozemohideen) [reply api/server/handlers/porter_app/create_and_update_events.go]: yessir

**Review by stefanmcshane — APPROVED** [review]

**Pull request merged**